signer/tests/integration/setup.rs

```diff
@@ -393,3 +399,431 @@ pub async fn backfill_bitcoin_blocks(db: &PgStore, rpc: &Client, chain_tip: &bit
         block_header = rpc.get_block_header_info(&parent_header_hash).unwrap();
     }
 }
+
+/// The information about a sweep transaction that has been confirmed.
+pub struct TestSignerSet {
+    /// The signer object. It's public key represents the group of signers'
+    /// public keys, allowing us to abstract away the fact that there are
+    /// many signers needed to sign a transaction.
+    pub signer: Recipient,
+    /// The public keys of the signer set. It is effectively controlled by
+    /// the above signer's private key.
+    pub keys: Vec<PublicKey>,
+}
+
+impl TestSignerSet {
+    pub fn new<R>(rng: &mut R) -> Self
+    where
+        R: rand::Rng,
+    {
+        let signer = Recipient::new(AddressType::P2tr);
+        let keys = signer::testing::wallet::create_signers_keys(rng, &signer, 7);
+        Self { signer, keys }
+    }
+
+    pub fn signer_keys(&self) -> &[PublicKey] {
+        &self.keys
+    }
+}
+
+/// The information about a sweep transaction that has been confirmed.
+#[derive(Debug)]
+pub struct SweepTxInfo {
+    /// The block hash of the bitcoin block that confirmed the sweep
+    /// transaction.
+    pub block_hash: BitcoinBlockHash,
+    /// The height of the bitcoin block that confirmed the sweep
+    /// transaction.
+    pub block_height: u64,
+    /// The transaction that swept in the deposit transaction.
+    pub tx_info: BitcoinTxInfo,
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct DepositAmounts {
+    pub amount: u64,
+    pub max_fee: u64,
+}
+
+/// A struct containing an actual deposit and a sweep transaction. The
+/// sweep transaction was signed with the `signer` field's public key.
+pub struct TestSweepSetup2 {
```

Comment:
Will this struct completely replace `TestSweepSetup` or should we give it a more descriptive name to differenciate the two?

Reply:
I want it to completely replace the original `TestSweepSetup`. But doing so is a (small) task so I left that fight for another day.